shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -37,121 +35,92 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
 
         public static string GetTypeDisplayName(Type type, bool fullName = true)
         {
-            var sb = new StringBuilder();
-            ProcessTypeName(type, sb, fullName);
-            return sb.ToString();
-        }
-
-        private static void AppendGenericArguments(Type[] args, int startIndex, int numberOfArgsToAppend, StringBuilder sb, bool fullName)
-        {
-            var totalArgs = args.Length;
-            if (totalArgs >= startIndex + numberOfArgsToAppend)
-            {
-                sb.Append("<");
-                for (int i = startIndex; i < startIndex + numberOfArgsToAppend; i++)
-                {
-                    ProcessTypeName(args[i], sb, fullName);
-                    if (i + 1 < startIndex + numberOfArgsToAppend)
-                    {
-                        sb.Append(", ");
-                    }
-                }
-                sb.Append(">");
-            }
+            var builder = new StringBuilder();
+            ProcessType(builder, type, fullName);
+            return builder.ToString();
         }
 
-        private static void ProcessTypeName(Type t, StringBuilder sb, bool fullName)
+        private static void ProcessType(StringBuilder builder, Type type, bool fullName)
         {
-            var typeInfo = t.GetTypeInfo();
-
-            if (typeInfo.IsGenericType)
+            if (type.IsGenericType)
             {
-                ProcessNestedGenericTypes(t, sb, fullName);
+                var genericArguments = type.GetGenericArguments();
+                ProcessGenericType(builder, type, genericArguments, genericArguments.Length, fullName);
             }
-            else if (typeInfo.IsArray)
+            else if (type.IsArray)
             {
-                ProcessArrayTypes(t, sb, fullName);
+                ProcessArrayType(builder, type, fullName);
             }
-            else if (_builtInTypeNames.ContainsKey(t))
+            else if (_builtInTypeNames.TryGetValue(type, out var buildInName))
```

Comment:
The `out var` should be named `builtInName`.

Reply:
Sorry for the typo, and thanks for finding it,  I'll fix it in one of the future PR's.

Reply:
😄